dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/gdv/custom/GdvYearlyDecimalTimeseriesDataComponent.kt

```diff
@@ -0,0 +1,121 @@
+package org.dataland.frameworktoolbox.frameworks.gdv.custom
+
+import com.fasterxml.jackson.core.json.JsonWriteFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.addStandardCellWithValueGetterFactory
+import org.dataland.frameworktoolbox.intermediate.components.support.SelectionOption
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.SectionUploadConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.capitalizeEn
+
+/**
+ * A GdvYearlyDecimalTimeseriesDataComponent is an in-memory representation of a generic field
+ * that encodes several values across a span of multi years. It is displayed / upload in a matrix
+ */
+class GdvYearlyDecimalTimeseriesDataComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    /**
+     * TODO
+     */
+    data class TimeseriesRow(val identifier: String, val label: String, val unitSuffix: String)
+
+    var decimalRows: MutableList<TimeseriesRow> = mutableListOf()
```

Comment:
What is this decimalrow?

Reply:
one row of fields (3 before and after the reporting period). its what you see on the view and upload page.

Reply:
what we discussed in the ui ux

Reply:
You mean the different rows in the rolling window component?
The name is a bit confusion, because i would have thought it is something like rows in the data dictionary file.

Reply:
Exactly. This property contains a list of rows in the rolling window component. I do not have any strong feeling regarding naming.

Reply:
i will leave it open for now, if a good suggestion comes to my mind, if not then i will close it at the end of the review

Reply:
As the rows hold BigDecimals, iam fine with the name and will close this